helpers/embeds.js

```diff
@@ -0,0 +1,65 @@
+import { html, nothing, render } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { map } from 'lit/directives/map.js';
+import { tryGetPluginByKey } from '@brightspace-ui/core/helpers/plugins.js';
+
+const embedTypeAttributeName = 'data-d2l-embed-type';
+
+export async function createEmbedPlaceholder(embedType, options) {
+	const embedRendererPlugin = tryGetPluginByKey('d2l-html-embed-renderer', embedType);
+	if (!embedRendererPlugin) return;
+
+	const placeholderData = await embedRendererPlugin.getPlaceholder(options);
+	const contents = placeholderData.contents
+		? map(Object.entries(placeholderData.contents), ([id, content]) => html`<template data-d2l-embed-template-id=${id}>${content}</template>`)
+		: nothing;
+
+	const attributes = placeholderData.attributes ? JSON.stringify(placeholderData.attributes) : undefined;
+
+	return placeholderData.inline
+		? html`
+			<span data-d2l-embed-type="${embedType}" data-d2l-embed-props="${ifDefined(attributes)}">
+				${contents}
+			</span>
+		` : html`
+			<div data-d2l-embed-type="${embedType}" data-d2l-embed-props="${ifDefined(attributes)}">
+				${contents}
+			</div>
+		`;
+}
```

Comment:
I haven't tested this placeholder code because it requires wiring stuff up in the HTML editor that's (frankly) too much of a pain for me to want to deal with right now. So conceptually this is what I want to do, but the code itself might be bad or broken.

Reply:
I've now confirmed that this works as expected, at least for the profile card case.